### PR TITLE
Feat: add multiple account report link to account references

### PR DESCRIPTION
**Pull request by jniles**
This PR implements a link from "account references" to the multiple account statement, auto-populating the data in the multiple account select input.  We can use this mechanism to auto-populate other reports as well (such as stock).  Here is what it looks like in action:

rQM9i2kLQM

We also need something like this to filter our balance sheet, though that is probably best left for a totally different report (https://github.com/IMA-WorldHealth/bhima/issues/5094).

### CODE REVIEW

File: server/controllers/finance/accounts/references.js [hunk 1]
```diff
@@ -288,13 +287,30 @@ async function lookupAccountReference(id) {
   return reference;
 }
 
+/**
+ * @function getAccountsForReferenceHTTP
+ *
+ * @description
+ * HTTP interface to compute the accounts associated with a particular reference.
+ */
+async function getAccountsForReferenceHTTP(req, res, next) {
```
Comment (mbayopanda) [inline server/controllers/finance/accounts/references.js:296]: Why did you add `HTTP` in the name of the function, is that a convention?

**Review by mbayopanda — APPROVED** [review]
LGTM 👍

File: server/controllers/finance/accounts/references.js (hunk 1 above)
Comment (jniles) [reply server/controllers/finance/accounts/references.js:296]: There is another function name almost the same thing here.  So, to make sure it wasn't confusing, I named this function that takes in `req`, `res`, `next` "HTTP".  It isn't really a convention - just lack of creativity on my part.

If you can think of a  better name, I would be open to it 😄

**Comment by jniles** [comment]
@mbayopanda can this be merged?
